conans/client/conf/config_installer.py

```diff
@@ -1,12 +1,34 @@
 import os
-from conans.tools import unzip
 import shutil
+from six.moves.urllib.parse import urlparse
+
+from conans.tools import unzip
 from conans.util.files import rmdir, mkdir
 from conans.client.remote_registry import RemoteRegistry
 from conans import tools
 from conans.errors import ConanException
 
 
+def _remove_credentials(resource):
+    """
+    Remove credentials from resource url/file path
+
+    :param resource: string with url or file path
+    :return: resource with credentials removed if present
+    """
+    if not resource:
```

Comment:
Still not nice. The result of passing a None resource, or an empty list, or empty whatever, is not to return that object.
A precondition of this function is that resource is a string. The function should fail if this is not the case. Not necessary to  do it explicitly, let the function code fail.